borg/hashindex.pyx

```diff
@@ -28,13 +28,24 @@ cdef extern from "_hashindex.c":
 
 cdef _NoDefault = object()
 
+"""
+The HashIndex is *not* a general purpose data structure. The value size must be at least 4 bytes, and these
+first bytes are used for in-band signalling in the data structure itself::
+
+    (uint32_t) -1 (= all bits set) mark deleted entries.
+    ((uint32_t) -1) - 1 mark empty entries.
```

Comment:
This probably should reference byte order right here. -1 is the same everywhere, kind of like 0, but -2 would have a different byte changed on LSB vs MSB.